Adding randomness for users, pwds, and hosts for datastore entries

eliserichards opened the pull request:
## Testing and Review Notes
This does not have an associated issue on our board. The purpose of this PR is to take some of the randomness functionality from [FxA's fake passwords](https://gist.github.com/rfk/916d9ca684f862b1c1030c685a5a4d19) in order to have better randomness in our test entries.

**Before**
<img src="https://user-images.githubusercontent.com/43795363/47679869-8184de00-db82-11e8-8270-49a72a361a1f.png" height="600"> <img src="https://user-images.githubusercontent.com/43795363/47679886-8a75af80-db82-11e8-8732-e0d28949a479.png" height="600">


**After**
<img src="https://user-images.githubusercontent.com/43795363/47679731-2f43bd00-db82-11e8-8896-7c637e3b1ce6.png" height="600"> <img src="https://user-images.githubusercontent.com/43795363/47679756-3e2a6f80-db82-11e8-933e-434eb0e7d3aa.png" height="600">

## To Do
- [x] double check the original issue to confirm it is fully satisfied - confirmed with @devinreams that it meets requirements for better usernames/passwords/hostnames
- [x] add testing notes and screenshots in PR description to help guide reviewers
- ~[ ] add unit tests~
  - optional: consider adding instrumentation (integration/UI) tests
- consider running this branch in a debug simulator and check for memory leak notifications or any warnings
- ~[ ] request the "UX" team perform a design review (if/when applicable)~
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)

linuxwolf commented on app/src/main/java/mozilla/lockbox/support/DataStoreSupport.kt:
```diff
@@ -42,25 +45,70 @@ class FixedDataStoreSupport(values: List<ServerPassword>? = null) : DataStoreSup
+    var hostnameLength = getRandomInRange(8, 20)
+    val hostname =
+        buildSequence {
+            val r = Random(); while (true) yield(r.nextInt(24))
```
related to "67 vs 65" below, why use `24` instead of `26`?

eliserichards replied:
same as other. changing this

linuxwolf commented on app/src/main/java/mozilla/lockbox/support/DataStoreSupport.kt:
```diff
@@ -42,25 +45,70 @@ class FixedDataStoreSupport(values: List<ServerPassword>? = null) : DataStoreSup
-            timeLastUsed = used,
-            timePasswordChanged = changed,
-            timesUsed = rng.nextInt(100) + 1
+        id = "0000$idx",
```
this ought to be `UUID.randomUUID().toString()` ... the original implementer (me) was too lazy (-:

eliserichards replied:
I'll add the UUID impl 👍

eliserichards commented on app/src/main/java/mozilla/lockbox/support/DataStoreSupport.kt:
```diff
@@ -42,25 +45,70 @@ class FixedDataStoreSupport(values: List<ServerPassword>? = null) : DataStoreSup
+        }
+            .take(hostnameLength)
+            .map {
+                (it + 67).toChar().toLowerCase()
```
good call! Also, 67/65 was a typo. Good catch 👍

eliserichards commented:
@linuxwolf - I addressed your comments. Also updated the navigation_version re: the linter error ("`...a newer version of android.arch.navigation:navigation-fragment than 1.0.0-alpha06 is available: 1.0.0-alpha07 [GradleDependency]`")
